disperser/controller/metadata/batch_metadata_manager.go

```diff
@@ -0,0 +1,199 @@
+type batchMetadataManager struct {
+	ctx    context.Context
+	logger logging.Logger
+
+	// The underlying eth client used to interact with the blockchain.
+	ethClient bind.ContractBackend
+
+	// Used to get operator state. The IndexedChainState utility fetches state both from onchain sources and from
+	// the indexer. When we eventually move all data onchain, we can ditch the indexer and just call directly
+	// into the contract bindings in this file.
+	indexedChainState core.IndexedChainState
+
+	// A handle for communicating with the registry coordinator contract.
+	registryCoordinator *regcoordinator.ContractRegistryCoordinator
+
+	// When choosing a new reference block number (RBN), select the block that is this many blocks in the past.
+	// This is a hedge against forking.
+	referenceBlockOffset uint64
+
+	// The time between updates to the metadata.
+	updatePeriod time.Duration
+
+	// The most recent batch metadata.
+	metadata atomic.Pointer[BatchMetadata]
+
+	alive atomic.Bool
+}
+
+// Create a new BatchMetadataManager with the specified quorums.
+//
+// updatePeriod is the period at which the manager will update its metadata. Actual update timing may vary
+// depending on the amount of time it takes to successfully get new data.
+func NewBatchMetadataManager(
+	ctx context.Context,
+	logger logging.Logger,
+	ethClient bind.ContractBackend,
+	indexedChainState core.IndexedChainState,
+	registryCoordinatorAddress gethcommon.Address,
+	updatePeriod time.Duration,
+	referenceBlockOffset uint64,
+) (BatchMetadataManager, error) {
+
+	registryCoordinator, err := regcoordinator.NewContractRegistryCoordinator(registryCoordinatorAddress, ethClient)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create registry coordinator client: %w", err)
+	}
+
+	manager := &batchMetadataManager{
+		ctx:                  ctx,
+		logger:               logger,
+		ethClient:            ethClient,
+		metadata:             atomic.Pointer[BatchMetadata]{},
+		indexedChainState:    indexedChainState,
+		registryCoordinator:  registryCoordinator,
+		referenceBlockOffset: referenceBlockOffset,
+		updatePeriod:         updatePeriod,
+	}
+	manager.alive.Store(true)
+
+	// Make sure we have valid metadata before the constructor returns.
+	err = manager.updateMetadata()
+	if err != nil {
+		return nil, fmt.Errorf("failed to update initial metadata: %w", err)
+	}
+
+	go manager.updateLoop()
+
+	return manager, nil
+}
+
+// GetMetadata returns the most recent batch metadata. This method is thread safe.
+func (m *batchMetadataManager) GetMetadata() *BatchMetadata {
+	return m.metadata.Load()
+}
+
+// Close releases resources associated with this manager.
+func (m *batchMetadataManager) Close() {
+	m.alive.Store(false)
+}
+
+// Fetch the next reference block number (RBN) to use.
+func (m *batchMetadataManager) getNextReferenceBlockNumber() (uint64, error) {
+	// Get the latest block header to determine the current reference block number.
```

Comment:
> the current reference block number

I don't think it makes sense to talk about the "current reference block number". This is just the "current block number", and from that we *choose* an RBN

Reply:
```
	// Get the latest block header to find the latest block number, 
	// which we use to choose a good current reference block number.
```